networking_dvs/utils/dvs_util.py

```diff
@@ -445,6 +436,23 @@ def create_dvportgroup(self, sg_attr_key, sg_set, port_config):
         except vmware_exceptions.VimException as e:
             raise exceptions.wrap_wmvare_vim_exception(e)
 
+    def dvportgroup_name(self, sg_set):
+        """
+        Returns a dvportgroup name for the particular security group set
+        in the context of the current switch
+        """
+        # There is an upper limit on managed object names in vCenter
+        dvs_id = ''.join(self.uuid.split(' '))[:8]
+        name = dvs_id + "-" + sg_set
+        if len(name) > 80:
+            # so we use a hash of the security group set
+            hex = hashlib.sha224()
+            hex.update(dvs_id)
```

Comment:
And maybe remove the `dvs_id` from the hash.
We have the suffix already in the name, and then the same set of security groups hashes to the same value. Which is a plus, in my mind.